libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,233 @@
+            services.AddBotRuntimeAdapters(runtimeSettings);
+        }
+
+        internal static void AddBotRuntimeSkills(this IServiceCollection services, SkillSettings skillSettings)
+        {
+            services.AddSingleton(sp => new AuthenticationConfiguration { ClaimsValidator = new AllowedCallersClaimsValidator(skillSettings?.AllowedCallers) });
+            services.AddSingleton<SkillConversationIdFactoryBase, SkillConversationIdFactory>();
+            services.AddHttpClient<BotFrameworkClient, SkillHttpClient>();
+            services.AddSingleton<ChannelServiceHandler, SkillHandler>();
+        }
+
+        internal static void AddBotRuntimeStorage(this IServiceCollection services, IConfiguration configuration, RuntimeSettings runtimeSettings)
+        {
+            services.AddSingleton<UserState>();
+            services.AddSingleton<ConversationState>();
+
+            // Cosmosdb
+            if (runtimeSettings?.Storage == nameof(CosmosDbPartitionedStorage))
+            {
+                var cosmosDbOptions = configuration?.GetSection(nameof(CosmosDbPartitionedStorage)).Get<CosmosDbPartitionedStorageOptions>();
+                services.AddSingleton<IStorage>(sp => new CosmosDbPartitionedStorage(cosmosDbOptions));
+            }
+
+            // Blob
+            else if (runtimeSettings?.Storage == nameof(BlobsStorage))
+            {
+                var blobOptions = configuration?.GetSection(nameof(BlobsStorage)).Get<BlobsStorageSettings>();
+                services.AddSingleton<IStorage>(sp => new BlobsStorage(blobOptions?.ConnectionString, blobOptions?.ContainerName));
+            }
+
+            // Default
+            else
+            {
+                // If no storage is configured, default to memory storage
+                services.AddSingleton<IStorage, MemoryStorage>();
+            }
+        }
+
+        internal static void AddBotRuntimePlugins(this IServiceCollection services, IConfiguration configuration, RuntimeSettings runtimeSettings)
+        {
+            using (IServiceScope serviceScope = services.BuildServiceProvider().CreateScope())
+            {
+                var pluginEnumerator = serviceScope.ServiceProvider.GetService<IBotPluginEnumerator>() ?? new AssemblyBotPluginEnumerator(AssemblyLoadContext.Default);
+
+                // Iterate through configured plugins and load each one
+                foreach (BotPluginDefinition plugin in runtimeSettings.Plugins)
+                {
+                    plugin.Load(pluginEnumerator, services, configuration);
+                }
+            }
+        }
+
+        internal static void AddBotRuntimeAdapters(this IServiceCollection services, RuntimeSettings runtimeSettings)
+        {
+            const string defaultRoute = "messages";
+
+            // CoreAdapter dependencies registration
+            services.AddSingleton<ICredentialProvider, ConfigurationCredentialProvider>();
```

Comment:
We don't need these here.

Reply:
We do for this release. Otherwise we get an ambiguous constructor exception because of the constructor overlap in BotFrameworkHttoAdapter. Our samples don't need it because they don't require an AuthenticationConfiguration instance. The runtime requires it to control skills .